lib/src/util/condition_scope_visitor.dart

```diff
@@ -129,10 +129,10 @@ abstract class ConditionScopeVisitor extends RecursiveAstVisitor {
   ConditionScope? outerScope;
   final breakScope = BreakScope();
 
-  Iterable<Expression?> getFalseExpressions(Iterable<Element?> elements) =>
+  Iterable<Expression?>? getFalseExpressions(Iterable<Element?> elements) =>
```

Comment:
Would it be reasonable to return an empty `Iterable` rather than `null` here?

Reply:
I think this would be a bit cleaner but with the current code flow would end up with some minor unnecessary objects created and logic running. Both ways seem fine, but perhaps this could be addressed afterwards.

Reply:
Exactly.  When I mentioned in the description above that there were a few places where I stopped short of a a bigger redesign this is one of them.  It would be great if we did a sweep at some point.  The are a bunch of places where we're returning nullable values where we could do a bit better.  (A nullable collection feels like an anti-pattern if you ask me.)

Reply:
I'll add a TODO.

Reply:
There are times when there's a legitimate need to distinguish between an empty collection and `null`, but I agree that when that's not the case it's better to return an empty collection than `null`.